source/jormungandr/jormungandr/scenarios/helper_classes/fallback_durations.py

```diff
@@ -152,8 +153,18 @@ def _do_request(self):
         else:
             origins = places_isochrone
             destinations = [center_isochrone]
-        sn_routing_matrix = self._instance.get_street_network_routing_matrix(
-            origins, destinations, self._mode, self._max_duration_to_pt, self._request, **self._speed_switcher
+
+        streetnetwork_service = self._instance.get_street_network(self._mode, self._request)
+        custom_event = new_relic.DistributedEvent(streetnetwork_service, "routing_matrix", "street_network")
```

Comment:
Don't know if decorators will do the job but yes, it's a bit heavy like this (but I like the functionnality).